### ISSUE TITLE
Add search contexts

### ISSUE THREAD

**Issue by christinaforney** [sourcegraph/sourcegraph#15396]
Search contexts let developers search just the code they care about.

<p></p>

For the first iteration, users cannot make changes to the default search contexts. In a later iteration, users will be able to create their own search contexts targeting a specific subset of repositories and/or a mix of existing search contexts, and graphs like *"...and their dependencies."*

<p></p>

When a user adds a repository, it's automatically added to the "My repos" search context.

<p></p>

**Action items for first iteration:**

<p></p>

- New view in user settings

- Search context in searches

<p></p>

**On the radar in case it affects implementation decisions:**

<p></p>

- Organizations will use the same search contexts UI in the organization's settings.

- Organizations will have a default search context for repositories they add. All members of that organization on Sourcegraph will have access to that search context.

- New organization members that join via an invite path will automatically use the organization's search context as the default, not "My repos."

- Dropdown UI will be revisited when organizations are implemented (name in dropdwon element may be too long).

- Users and organizations will eventually be able to create new search contexts.

**Comment by christinaforney** [comment]
Link to feature: https://sourcegraph.productboard.com/feature-board/planning/features/6031014

**Comment by tsenart** [comment]
@ryanslade, @quinnkeast: Does this issue overlap with any other you have created before?

**Comment by ryanslade** [comment]
No, but I believe some of the details might change after discussions with the search team.

@artemruts I'm going to add this to the Milestone 3 project, but we should't start work on it until we've had back from the search team.

**Comment by quinnkeast** [comment]
Please refer to #15623 instead of this issue. Some Productboard sync changes.
